src/goto-instrument/contracts/utils.h

```diff
@@ -112,17 +118,102 @@ void insert_before_swap_and_advance(
 /// \param mode: The mode for the new symbol, e.g. ID_C, ID_java.
 /// \param symtab: The symbol table to which the new symbol is to be added.
 /// \param suffix: Suffix to use to generate the unique name
-/// \param is_auxiliary: Do not print symbol in traces if true (default = false)
+/// \param is_auxiliary: Do not print symbol in traces if true (default = true)
 /// \return The new symbolt object.
 const symbolt &new_tmp_symbol(
   const typet &type,
   const source_locationt &location,
   const irep_idt &mode,
   symbol_table_baset &symtab,
   std::string suffix = "tmp_cc",
-  bool is_auxiliary = false);
+  bool is_auxiliary = true);
 
 /// Add disable pragmas for all pointer checks on the given location
 void disable_pointer_checks(source_locationt &source_location);
 
+/// Turns goto instructions `IF cond GOTO label` where the condition
+/// statically simplifies to `false` into SKIP instructions.
+void simplify_gotos(goto_programt &goto_program, namespacet &ns);
+
+/// Returns true iff the given program is loop-free,
+/// i.e. if each SCC of its CFG contains a single element.
+bool is_loop_free(
+  const goto_programt &goto_program,
+  namespacet &ns,
+  messaget &log);
+
+/// Stores information about a goto function computed from its CFG,
+/// together with a `target` iterator into the instructions of the function.
+///
+/// The methods of this class provide information about identifiers at
+/// the current `target` instruction to allow simplifying assigns
+/// clause checking assertions.
+class cfg_infot
+{
+public:
+  cfg_infot(const namespacet &_ns, goto_functiont &_goto_function)
+    : ns(_ns),
+      goto_function(_goto_function),
+      target(goto_function.body.instructions.begin()),
+      bitvector_analysis(goto_function, ns),
+      dirty_analysis(goto_function),
+      dirty_idents(dirty_analysis.get_dirty_ids()),
```

Comment:
Why do you copy these instead of using `dirty_analysis` when needed?

Reply:
Removed